net: eth: Remove old IPv6 interface identifier address

jukkar opened the pull request:
If we change the ethernet MAC address, then we must also remove
the old IPv6 interface identifier (iid) address from the
interface. Otherwise there might not be enough space in the IPv6
address array for the new iid address and beside the old iid
address is not useful any more after the MAC address is changed.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
net: eth: Remove old IPv6 interface identifier address

If we change the ethernet MAC address, then we must also remove
the old IPv6 interface identifier (iid) address from the
interface. Otherwise there might not be enough space in the IPv6
address array for the new iid address and beside the old iid
address is not useful any more after the MAC address is changed.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

rveerama1 commented on subsys/net/l2/ethernet/ethernet_mgmt.c:
```diff
@@ -88,6 +88,25 @@ static int ethernet_set_config(u32_t mgmt_request,
+			/* No need to check the return value in this case. It
+			 * is not an error in this case if the address is not
+			 * found.
+			 */
```
How about a warning or info message?

rveerama1 replied:
ok

jukkar commented on subsys/net/l2/ethernet/ethernet_mgmt.c:
```diff
@@ -88,6 +88,25 @@ static int ethernet_set_config(u32_t mgmt_request,
 			return -EACCES;
 		}
 
+#if defined(CONFIG_NET_IPV6)
```
Yes, we can do it like that too, just requires that we create the dummy funcs for those. I can add a patch in this set that creates those dummy ones.

jukkar committed:
net: if: Allow calling IPv6 specific functions

Make sure that IPv6 specific functions are callable even if
IPv6 is not enabled. This allows use of IS_ENABLED() macro
in other parts of the system.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
tests: net: ethernet_mgmt: Enable IPv6 for this test

IPv6 is needed as there is special handling of IPv6 addresses
when MAC address is changed. We do not need DAD + other IPv6
specific features so those are disabled.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar commented:
Added patch that allows call of various IPv6 specific functions even if IPv6 is disabled.

jukkar committed:
net: if: Allow calling IPv4 specific functions

Make sure that IPv4 specific functions are callable even if
IPv4 is not enabled. This allows use of IS_ENABLED() macro
in other parts of the system.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar commented:
Fixed sanity error when generating documentation.

jukkar commented:
Rebasing because of doc warning fixes.

jukkar commented:
recheck

pfalcon commented:
Re: "net: if: Allow calling IPv6 specific functions". IMHO, that's going more and more confusing, for someone who would just start to get acquainted with the code. OTOH, I fully understand/agree that it's practical and improves maintainability. Bottom line, that it should be described in some imaginable "Zephyr IP stack developer's primer" ;-).

pfalcon commented:
> This allows use of IS_ENABLED() macro
in other parts of the system.

Oh, now I got (see) it. The intention is not to call ipv6-specific functions without guards, it's just be able to use `if (IS_ENABLED())` guards in addition to the `#ifdef` ones. Ok. (Once again would mention that we should allow indented `#ifdef`s to alleviate that issue too.)
